### PR TITLE
docs: Update references for transparent proxy with peers

### PR DESCRIPTION
**Pull request by kisunji**
### Description
Did another pass through of docs re: tproxy.

One thing I noticed was that it's very hard to navigate from Transparent Proxy to any of the relevant information (service-registration and dns)

### CODE REVIEW

File: website/content/docs/discovery/dns.mdx [hunk 1]
```diff
@@ -488,18 +488,17 @@ fields must be present:
...
-For node lookups, only the partition and datacenter need to be specified as nodes cannot be
-namespaced.
+For imported lookups, only the namespace and peer need to be specified as the partition that is peered can be inferred:
 
 ```text
-[tag.]<service>.service.<partition>.ap.<datacenter>.dc.<domain>
+<service>.virtual[.namespace][.peer].<domain>
 ```
 
-For node lookups, only the partition and datacenter need to be specified (nodes cannot be
```
Comment (kisunji) [inline website/content/docs/discovery/dns.mdx:498]: The diff is misleading; there were similarly worded sections for node (which had the wrong copy-pasted example)

File: website/content/docs/connect/transparent-proxy.mdx [hunk 2]
```diff
@@ -29,8 +29,8 @@ Without transparent proxy, application owners need to:
 
 With transparent proxy:
 
-1. Upstreams are inferred from service intentions, so no explicit configuration
-   is needed.
+1. Upstreams are inferred from service intentions and/or imported services,
```
Comment (mkeeler) [inline website/content/docs/connect/transparent-proxy.mdx:32]: Should we clarify that the intentions we infer upstreams from are the ones from the local cluster and not ones set in remote peers?

Comment (kisunji) [reply website/content/docs/connect/transparent-proxy.mdx:32]: Yes, that's a good point.

**Review by trujillo-adam — APPROVED** [review]
lgtm

**Pull request merged**